### DISCUSSION TITLE
Declaring test suite-wide filterwarnings setting in code

### DISCUSSION THREAD

**Discussion by jwodder** [pytest-dev/pytest#9935] [Q&A]
We currently have a number of `[pytest]filterwarnings` settings defined for our project in `tox.ini`, but, for various reasons, we need these settings moved to the test code itself instead of being placed in a config file.  How can we do this?  Would setting `pytestmark` in `conftest.py` affect the whole test suite as desired?  Can/should this be done by calling `config.addinivalue_line` in `pytest_configure`?

**Comment by RonnyPfannschmidt** [comment]
the easiest way is probably to add markers for the warnings filters to the session object

**Reply by jwodder** [reply]
How exactly do I do that?  Can you provide some sample code?

**Reply by yarikoptic** [reply]
@RonnyPfannschmidt could you please share the how-to knowledge for this?

**Reply by RonnyPfannschmidt** [reply]
In a pytest_sessionstart hook you would do session.add_marker with the the filter warnings hooks

**Reply by jwodder** [reply]
Unfortunately, there are some warnings emitted during test collection (e.g., from deprecated modules imported by a dependency), and this approach doesn't silence them.  Is there another/better approach?

**Reply by RonnyPfannschmidt** [reply]
Invoking addinilines for filter warnings in a Pytest-configure hook

**Reply by jwodder** [reply]
That doesn't help.  I believe some of the warnings are emitted by modules indirectly imported by the package's `__init__.py`.

**Reply by RonnyPfannschmidt** [reply]
In that case without more context on what the code does, there is nothing to be done

**Reply by yarikoptic** [reply]
So the point seems to have some way for pytest while to test module X to load in configuration for X **before** importing module X. Is there anything in pytest which would parse or import otherwise some `X.conftest` without importing the `X` itself?
